lute/io/src/io.cpp

```diff
@@ -0,0 +1,141 @@
+#include "lute/io.h"
+#include "Luau/Variant.h"
+#include "lute/runtime.h"
+
+#include <uv.h>
+#include <memory>
+
+#include "lua.h"
+#include "lualib.h"
+
+
+namespace io
+{
+
+struct IOHandle
+{
+    Luau::Variant<uv_pipe_t, uv_tty_t> streamVariant;
```

Comment:
Some options here - asking you since I actually don't know the details.
The libuv docs say that `uv_pipe_t/uv_tty_t` are both 'subtypes' of `uv_stream_t`. Can we replace this variant with just that - is there a reason you care about which kind of stream this is?

Option 2 - can this be generic? Then you can instantiate the `IOHandle<uv_pipe_t>` and name it with a `using` statement.

The reason I ask is I don't see where you use the streamVariant in this structure, besides storage.